pkg/antctl/raw/multicluster/common/common.go

```diff
@@ -431,3 +447,52 @@ func newServiceAccount(name string, namespace string) *corev1.ServiceAccount {
 		},
 	}
 }
+
+func OutputJoinConfig(cmd *cobra.Command, file *os.File, clusterSetID, leaderClusterID, leaderNamespace string) error {
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+
+	config := &ClusterSetJoinConfig{
+		APIVersion:      ClusterSetJoinConfigAPIVersion,
+		Kind:            ClusterSetJoinConfigKind,
+		Namespace:       "",
+		ClusterID:       "",
+		LeaderClusterID: leaderClusterID,
+		LeaderAPIServer: kubeconfig.Host,
+		LeaderNamespace: leaderNamespace,
+		ClusterSetID:    clusterSetID,
+	}
+
+	var writeErr error
+	defer func() {
+		if writeErr != nil {
+			fmt.Fprintf(cmd.OutOrStderr(), "Error writing to %s: %v\n", file.Name(), writeErr)
+		}
+	}()
+
+	b, _ := yaml.Marshal(config)
+	if _, writeErr = file.Write([]byte("---\n")); writeErr != nil {
+		return writeErr
+	}
+	if _, writeErr = file.Write(b); writeErr != nil {
+		return writeErr
+	}
+
+	// We comment out these ClusterSetJoinConfig fields in the generated
+	// join config file, so they can be populated by command line options of
+	// the "antctl mc join" command.
+	optionalFields := `#clusterID: ""
+#namespace: ""
+# Use the pre-created token Secret.
+#tokenSecretName: ""
+# Create a token Secret with the manifest file.
+#tokenSecretFile: ""
+`
+	if _, writeErr = file.Write([]byte(optionalFields)); writeErr != nil {
```

Comment:
Seems the output is not what we expected:
The field name is started with upper case. @jianjuns  you may need unit tests to cover this.
```
---
APIVersion: multicluster.antrea.io/v1alpha1
ClusterID: ""
ClusterSetID: test-clusterset
Kind: ClusterSetJoinConfig
LeaderAPIServer: https://10.176.27.224:6443
LeaderClusterID: test-cluster-east
LeaderNamespace: antrea-multicluster
Namespace: ""
TokenSecretFile: ""
TokenSecretName: ""
#clusterID: ""
#namespace: ""
# Use the pre-created token Secret.
#tokenSecretName: ""
# Create a token Secret with the manifest file.
#tokenSecretFile: ""
```
cc @hjiajing , can you help to check if it's existing code issue or not? thanks.

Reply:
I did not pay attention to this in my tests. It is due to we used the yaml package from sigs.k8s.io. Already fixed. I am still adding more UT.